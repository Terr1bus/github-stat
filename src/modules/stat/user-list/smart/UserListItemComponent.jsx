import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const User = styled.li`
  display: flex;
  align-items: center;
  padding: 0 10px;
  margin-top: 5px;
  background: inherit;
  color: #fff;
  transition: .4s ease;

  &:hover {
    background: #ccc;
  }

  & > span {
    margin-left: 15px;
  }

  & > a {
    margin-left: auto;
    color: inherit;
    height: 100%;
  }
`;

const Avatar = styled.img`
  height: 30px;
  border-radius: 50%;
`;



const UserListItemComponent = ({ avatar_url, html_url, login }) => (
  <User>
    <Avatar src={avatar_url} alt="avatar" />
    <Link to={`/users/${login}`}>{login}</Link>
    <a href={html_url} target='_blank' rel='noopener noreferrer'>
      Github profile
    </a>
  </User>
);

UserListItemComponent.propTypes = {
  avatar_url: propTypes.string.isRequired,
  login: propTypes.string.isRequired,
  html_url: propTypes.string.isRequired,
};

export default UserListItemComponent;