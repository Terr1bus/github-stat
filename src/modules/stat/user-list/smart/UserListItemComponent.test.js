import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import UserListItemComponent from './UserListItemComponent';

afterEach(cleanup);

const props = {
  avatar_url: 'https://someing.com',
  html_url: 'https://somesite.com',
  login: 'login',
};

const getComponent = ({ avatar_url, login, html_url } = props) => (
  render (
    <MemoryRouter>
      <UserListItemComponent
        login={login}
        html_url={html_url}
        avatar_url={avatar_url}
      />
    </MemoryRouter>
  )
);

describe('<UserListItemComponent /> spec', () => {
  test('should match snapshot', () => {
    const { container } = getComponent();
    expect(container).toMatchSnapshot();
  });

  test('should render img with src from props', () => {
    const { container } = getComponent();
    const img = container.querySelector('img');
    expect(img).toHaveAttribute('src', props.avatar_url);
    expect(img).toHaveAttribute('alt', 'avatar');
  });

  test('should render <Link /> with href from props', () => {
    const { getByText } = getComponent();
    const link = getByText(props.login);
    expect(link).toHaveAttribute('href', `/users/${props.login}`);
  });

  test('should render link with href to address from props', () => {
    const { getByText } = getComponent();
    const githubLink = getByText('Github profile');
    expect(githubLink).toBeInTheDocument();
    expect(githubLink).toHaveAttribute('href', props.html_url);
  });
});