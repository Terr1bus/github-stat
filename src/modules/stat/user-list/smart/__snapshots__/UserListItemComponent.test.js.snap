// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserListItemComponent /> spec should match snapshot 1`] = `
<div>
  <li
    class="sc-bdVaJa eWpYVg"
  >
    <img
      alt="avatar"
      class="sc-bwzfXH cBIOLc"
      src="https://someing.com"
    />
    <a
      href="/users/login"
    >
      login
    </a>
    <a
      href="https://somesite.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Github profile
    </a>
  </li>
</div>
`;
